### DISCUSSION TITLE
How to get more granular lead/ gap time data.

### DISCUSSION THREAD

**Discussion by timf34** [theOehrly/Fast-F1#241] [Ideas]
Is there a way to get more granular lead/ gap time data between the cars. From what I can currently see, there is only timing data for each car at the start/ end of the race, and then at each sector, and then from this you can infer the leadTime between the leading car, and the cars after it (for example car 2 is 1.2 seconds behind the leading car 1).

Is it possible to get this data through the dataset? I can't seem to see it/ see how 

Thanks!

**Comment by theOehrly** [comment]
The API does provide the data more granularly (gap to leader only). It's not properly implemented/tied into the rest of FastF1, though. 
The function `fastf1.api.timing_data` returns `laps_data` and `stream_data`. The "stream_data" is what you are looking for.

**Reply by timf34** [reply]
Fantastic - thank you!!!
